dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemory.cs

```diff
@@ -134,7 +134,7 @@ public AzureCognitiveSearchMemory(string endpoint, TokenCredential credentials)
         string collection,
         string query,
         int limit = 1,
-        double minRelevanceScore = 0.7,
+        double minRelevanceScore = 0.0,
```

Comment:
this changes the behavior from returning something relevant to returning anything, e.g. potentially irrelevant information that will affect the quality of every component using this. E.g. the planner searching for relevant skills will get irrelevant ones in many more scenarios. I would be more useful having a default behavior that exclude some noise, e.g. in this case 0.5.

Reply:
It will always return the 'most relevant' result(s). In some cases, the 'most relevant' result may have a similarity score of <0.5. For example, if you're using sentence-transformers/all-MiniLM-L6-v2, a similarity score of 0.35 is quite useful, but for text-similarity-ada-002, this score would be considered not very similar. Hardcoding a threshold is very likely to return empty, which is not expected when searching for a 'nearest' value.

Reply:
FWIW ActionPlanner doesn't use relevancy and for SequentialPlanner it is off by default and users must supply a value here of their choice to enable relevancy filtering.

Reply:
@awharrison-28 "It will always return the 'most relevant' result(s)." sure, but when the most relevant has a similarity of 0.1 it's hardly useful and can lead to hallucination and/or incorrect text generation. This is such a core component that a similar change will propagate in multiple places. I would prefer a safer a approach, at the cost of not returning any data. Users should override the value to take "more risk" (decreasing the default value) rather than taking "less risk" (increasing the threshold).